### ISSUE TITLE
WYSIWYG image upload dialog hangs up

### ISSUE THREAD

**Issue by sshymko** [magento/magento2#23966]
WYSIWYG editor in the admin panel _sometimes_ hangs up upon opening an image upload dialog. The "Please wait..." spinner animation does not go away (until you refresh the page and have more luck next time).

### Preconditions (*)
1. Magento 2.3.1 Open Source
2. Google Chrome 74.0.3729.169 (Official Build) (64-bit)
2. Configuration > General > Content Management > WYSIWYG Options > Enable WYSIWYG Editor: Enabled by Default

### Steps to reproduce (*)
1. Log in to the admin panel
2. Go to Content > Elements > Pages
3. Edit any page, such as Home page
4. Expand Content section
5. Click on Insert/edit image toolbar button
6. Click on Browse button of the Source field
7. _Refresh the page and repeat until reproduced_

### Expected result (*)
1. Dialog "Select Images" shows up fully operational

### Actual result (*)
1. Dialog "Select Images" shows up with "Please wait..." infinite spinner animation
    <img width="1182" alt="magento2_admin_cms_wysiwyg_img_upload_spinner" src="https://user-images.githubusercontent.com/1231423/62257160-070b1800-b3b9-11e9-8678-23475c1446f4.png">

2. Refreshing the page and repeating the steps sometimes fixes the issue

3. There're no JavaScript errors in the developer console

**Comment by sshymko** [comment]
@magento give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @sshymko. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @sshymko, here is your [Magento instance](https://i-23966-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-23966-2-3-develop.instances.magento-community.engineering/admin](https://i-23966-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by sshymko** [comment]
Confirmed the issue is reproducible on the vanilla `2.3-develop` instance provided above.
It took multiple page refreshes ~5-10 to reproduce the issue.
<img width="1179" alt="magento2_admin_cms_wysiwyg_img_upload_hang_up" src="https://user-images.githubusercontent.com/1231423/62257590-f65ba180-b3ba-11e9-853d-efd414b8fea7.png">

**Comment by sshymko** [comment]
The bug causes instability of the following MFTF tests:
```
E AdminAddImageToWYSIWYGBlockTestCest: Admin add image to wysiwyg block test (64.08s)
E AdminAddImageToWYSIWYGCMSTestCest: Admin add image to wysiwygcms test (36.67s)
E AdminAddImageToWYSIWYGCatalogTestCest: Admin add image to wysiwyg catalog test (34.74s)
E AdminAddImageToWYSIWYGNewsletterTestCest: Admin add image to wysiwyg newsletter test (32.48s)
```
```
[Facebook\WebDriver\Exception\TimeOutException]
```

The image upload issue affects all instances of WYSIWYG editor in the admin panel.

**Comment by krishprakash** [comment]
@magento give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @krishprakash. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @krishprakash, here is your [Magento instance](https://i-23966-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-23966-2-3-develop.instances.magento-community.engineering/admin](https://i-23966-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by krishprakash** [comment]
Hi @sshymko,
I'm not able to reproduce this issue as provided steps by you.
Is there any specific case in which you phase this issue ?

**Comment by sshymko** [comment]
@krishprakash 
The provided steps are precisely exact steps. The issue is random and does not always reproduce. It took me 5-10 page refreshes to reproduce it on the provided instance (the screenshot above). Could you please try to refresh the page and repeat the steps as many times as you can?

**Comment by ju2kr** [comment]
@sshymko I have exactly the same problem, it occurs at irregular intervals. So far I haven't been able to see a fixed pattern to reproduce the problem.

**Comment by sshymko** [comment]
The issue may be caused by race condition of JavaScript callbacks toggling "Please wait..." spinner.
Occurrence of the issue depends on the timing of AJAX calls and alignment of stars :)

**Comment by krishprakash** [comment]
@sshymko @ju2kr  I have tried several times but couldn't reproduce the issue.
It may be browser specific issue.
Can you please share the browser detail in which you produce this issue.
Thanks

**Comment by sshymko** [comment]
@krishprakash it's not enough to try 2-3 times. It literally takes 10 or sometimes more test case repetitions with the page refresh (Ctrl+R) in between. See the exact browser version in the preconditions.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @krishprakash 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-18861` were created

**Issue Available**: @krishprakash, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by konarshankar07** [comment]
Hello @sshymko ,
I think this issue might be fixed in https://github.com/magento/magento2/pull/23621. Can you please test with the mention pull request

Thanks

**Comment by sshymko** [comment]
@konarshankar07 
I can confirm #23621 fixes this issue. The bug is no longer reproducible with the patch. Thanks!

**Comment by konarshankar07** [comment]
Hello @sshymko 
Please close this issue if you are not getting this issue any more
Thanks

**Comment by sshymko** [comment]
@konarshankar07 
I can no longer reproduce the issue manually, but the corresponding automated MFTF tests still intermittently fail although less frequently than before. I don't think the issue is fully resolved.

**Comment by konarshankar07** [comment]
@sshymko 
Can you please provide me which MFTF testing you are getting the error?
Thanks

**Comment by sshymko** [comment]
@konarshankar07 
Failing MFTF tests are listed in my comment above https://github.com/magento/magento2/issues/23966#issuecomment-517079435

**Comment by engcom-Foxtrot** [comment]
@sshymko can you please provide detailed steps to reproduce test failures?

**Comment by sshymko** [comment]
@engcom-Foxtrot 

**Preconditions:**
- Local development environment (NOT pipeline of Magento Inc.)
- Selenium Server Standalone 3.141.5
- ChromeDriver 74.0.3729.6
- Google Chrome 74.0.3729.169
- Magento 2.3.x OS installed from scratch
    - Production mode
    - DI compiled & static generated
    - Static files deployed
    - Full Page Cache (FPC) disabled (or enabled w/ patch [#3937](https://github.com/magento/magento2ce/pull/3937))
    - Fix [#23621](https://github.com/magento/magento2/pull/23621) applied
    - [MFTF Requirements](https://devdocs.magento.com/mftf/docs/getting-started.html) fulfilled
        ```bash
        php bin/magento config:set cms/wysiwyg/enabled disabled
        php bin/magento config:set admin/security/admin_account_sharing 0
        php bin/magento config:set admin/security/use_form_key 0
        ```

**Steps to reproduce:**
1. Reindex everything & flush cache:
    ```bash
    php bin/magento indexer:reindex
    php bin/magento cache:flush
    ```
2. Start the Selenium server:
    ```bash
    java -jar selenium-server-standalone-3.141.5.jar
    ```
3. Generate entire MFTF test suite:
    ```bash
    vendor/bin/mftf generate:tests
    ```
4. Run the test suite (NOT in parallel):
    ```bash
    vendor/bin/codecept run functional -c dev/tests/acceptance/codeception.yml
    ```
5. Wait ~7h for the tests to finish
6. Repeat the steps as necessary

**Actual result:**
- Number of tests fail, among which random failures related to the WYSIWYG instability:
    ```
    Tests: 473, Assertions: 4323, Errors: 7, Failures: 31, Skipped: 26.
    ```
- Re-installing the app and rerunning the tests results in slightly different +/-5 failures

**Comment by magento-engcom-team** [comment]
Hi @sshymko. Thank you for your report. 
The issue has been fixed in magento/magento2#24333 by @p-bystritsky in 2.3-develop branch
Related commit(s):
 - [3acfc94d0d6d0dfa544c2317d26daab5978e9193](https://github.com/magento/magento2/commit/3acfc94d0d6d0dfa544c2317d26daab5978e9193)
 - [48b48dd46f09762fab3dff6f81d12969070a8f00](https://github.com/magento/magento2/commit/48b48dd46f09762fab3dff6f81d12969070a8f00)

The fix will be available with the upcoming 2.3.4 release.

**Comment by VladimirZaets** [comment]
Hi @sshymko. Thank you for your report. 
The issue has been fixed in magento/magento2#25556 by @engcom-Golf in 2.3-develop branch
Related commit(s):
 - [f22905662aeda6a0f2c073d650326dd6c2a6fd93](https://github.com/magento/magento2/commit/f22905662aeda6a0f2c073d650326dd6c2a6fd93)
 - [a07627c9b0c8dcb600cc1109a65c5a08ab375a71](https://github.com/magento/magento2/commit/a07627c9b0c8dcb600cc1109a65c5a08ab375a71)
 - [8e39f49b48692f2e820066815244e8532c247fde](https://github.com/magento/magento2/commit/8e39f49b48692f2e820066815244e8532c247fde)
 - [3be5ac8e2e83415b8cafc340486b298038ed0421](https://github.com/magento/magento2/commit/3be5ac8e2e83415b8cafc340486b298038ed0421)
 - [76a866ec70987dfb0e1c778db472a06c6d738781](https://github.com/magento/magento2/commit/76a866ec70987dfb0e1c778db472a06c6d738781)
 - [26dca1007e2458e29419ae5ecf7b77e0c1560658](https://github.com/magento/magento2/commit/26dca1007e2458e29419ae5ecf7b77e0c1560658)

The fix will be available with the upcoming 2.3.5 release.
